internal/configs/virtualserver.go

```diff
@@ -258,6 +258,11 @@ func upstreamHasKeepalive(upstream conf_v1alpha1.Upstream, cfgParams *ConfigPara
 }
 
 func generateLocation(path string, upstreamName string, upstream conf_v1alpha1.Upstream, cfgParams *ConfigParams) version2.Location {
+	proxyPassProtocol := "http"
```

Comment:
I suggest having a function like `generateProxyPassProtocol` that accepts an upstream as an argument. I suggest writing a unit test for that function.

Reply:
I forgot to add tests yes. I don't see a reason to split 3 lines of code into a different function for now.